YACReader/render.cpp

```diff
@@ -695,13 +695,14 @@ void Render::createComic(const QString &path)
     pagesEmited.clear();
 
     if (comic != nullptr) {
-        //comic->moveToThread(QApplication::instance()->thread());
         comic->invalidate();
-
+        // Dispatch pending events to guard against race conditons
+        // This needs to run before disconnect to clear queued signals
+        // while comic still is valid
+        QCoreApplication::sendPostedEvents(this);
         comic->disconnect();
```

Comment:
I think there is still an API race between the `QCoreApplication::sendPostedEvents(this)` and `comic->disconnect()` lines. The `comic`'s thread can post a signal-emitted event before the disconnection.

Quote from https://woboq.com/blog/how-qt-signals-slots-work-part3-queuedconnection.html:
> When posting an event (in QCoreApplication::postEvent), the event will be pushed in a per-thread queue (QThreadData::postEventList). The event queued is protected by a mutex, so there is no race conditions when threads push events to another thread's event queue.

> I also tested running the processing step after disconnect, but that produced stray signals.

What stray signals and why? I think reordering these two statements would eliminate the race...